### PR TITLE
Build upgrade and Release 0.20.0

### PR DESCRIPTION
**Pull request by juanpedromoreno**
This PR upgrades the build to the latest version of `cats`, `cats-effect`, `circe`, etc. Additionally, it ignores a random test failure.

Releases 0.20.0.

### CODE REVIEW

File: github4s/shared/src/test/scala/github4s/integration/GHIssuesSpec.scala [hunk 1]
```diff
@@ -36,7 +35,7 @@ trait GHIssuesSpec[T] extends BaseIntegrationSpec[T] {
     })
   }
 
-  "Issues >> Get" should "return an issue which is a PR" in {
+  "Issues >> Get" should "return an issue which is a PR" ignore {
```
Comment (fedefernandez) [inline github4s/shared/src/test/scala/github4s/integration/GHIssuesSpec.scala]: Should we create an issue for this?

Comment (juanpedromoreno) [reply github4s/shared/src/test/scala/github4s/integration/GHIssuesSpec.scala]: I think it's related to https://github.com/47deg/github4s/issues/233

**Comment by juanpedromoreno** [comment]
@BenFradet @fedefernandez I'm requesting again your review since I've added more changes in order to complete the Scalajs upgrade.

**Comment by aleksandr-vin** [comment]
Can you upload the artifacts to maven, please?

**Comment by juanpedromoreno** [comment]
They should be there, that PR deployed artifacts automatically ;)

**Comment by aleksandr-vin** [comment]
Now I see them. Previously they were missing. Thanks

**Pull request merged**